dnf/package.py

```diff
@@ -82,6 +81,11 @@ def _from_repo(self):
             return '@' + pkgrepo
         return self.reponame
```

Comment:
I know this is not part of your PR but I think the method should be straightened a bit (especially I do not understand using of the empty dictionary instead of None). But I'm not blocking the PR on this.
```
if self._from_system:
    pkgrepo = self.base.history.repo(self)
    if pkgrepo:
        return '@' + pkgrepo
return self.reponame
```

Reply:
I removed else part instead. Your suggestion was correct, it does no require a dict there, but an object that is recognized by puthon as an empty. Like "", None, (), ...